frontend/appflowy_flutter/integration_test/desktop/grid/grid_filter_and_sort_test.dart

```diff
@@ -0,0 +1,121 @@
+import 'package:appflowy/plugins/database/grid/presentation/widgets/filter/choicechip/checkbox.dart';
+import 'package:appflowy_backend/protobuf/flowy-database2/protobuf.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:integration_test/integration_test.dart';
+
+import '../../shared/database_test_op.dart';
+import 'grid_test_extensions.dart';
+
+void main() {
+  IntegrationTestWidgetsFlutterBinding.ensureInitialized();
+
+  group('grid simulatneous sort and filter test:', () {
+    testWidgets('delete filter with active sort', (tester) async {
+      await tester.openTestDatabase(v069GridFileName);
+
+      // get grid data
+      final original = tester.getGridRows();
+
+      // add a filter
+      await tester.tapDatabaseFilterButton();
+      await tester.tapCreateFilterByFieldType(
+        FieldType.Checkbox,
+        'Registration Complete',
+      );
```

Comment:
Failure caused by incorrect refresh of the frontend, leading to the new row being inserted at the wrong index. Refreshing the database view shows the correct row order.

When the new visible rows are passed, their indices are  larger than `_rowList.length`, so they are just appended randomly. The Tauri version solved this by re-fetching rows when this occurs